### ISSUE TITLE
Faster localization

### ISSUE THREAD

**Issue by matthewearl** [matthewearl/deep-anpr#5]
The current windowing scheme makes the detection process rather slow. The main reason for this is the high overlap between windows: Each pixel is covered at least 128 times at each scale, and there are a large number of scales processed due to the conservative scaling factor of root 2.

Inspired by [Joseph Redmon et al.'s YOLO paper](http://arxiv.org/abs/1506.02640) I propose applying a neural network in windows with no overlap within a particular scale (ie. a disjoint tiling), and with a larger scaling factor.   The network output would be a regression similar to the YOLO bounding box predictor.  The predicted bounding boxes would then be passed to the existing network in order to read the numbers.

**Comment by quanghoc** [comment]
Hi,

How long is the `train.py` supposed to run? It has been 4 days for me.

**Comment by SunAriesCN** [comment]
@quanghoc you can read the README.md about your problem.

**Comment by quanghoc** [comment]
It said 100K batches but I cannot see the batch count anywhere in the console.

**Comment by matthewearl** [comment]
@quanghoc The batch count should be visible near the bottom of the output, on the line starting B<number>.   <number> is the current batch number.  If you don't see this can you please raise a separate issue? Also note, this issue is for speeding up detect.py with a fast localization stage so I'll be deleting these comments shortly (general comments about speed should go in #3).

**Comment by matthewearl** [comment]
I've changed gen.py to output detection images and reading images.  The detection images have an expected output of `x,y,s` where `x,y` is the location of the centre of the plate, normalized to the [0, 1] interval and `s` is the size of the bounding box that most tightly fits the output plate. The bounding box has the same aspect ratio as the image, and is expressed as a scale of the original image dimensions.
 
The idea is a network architecture would be constructed which predicts these 3 values given the input image.

The code for this is in the `detection-regressor` branch.  For the purposes of experimentation, the gen module can be used independently of the rest of the existing architecture.  If anyone does any experimentation in this direction it would be good to hear about the results, even if they are negative.

**Comment by zomco** [comment]
Is it possible to combine localization and recognition into YOLO network?

**Comment by SunAriesCN** [comment]
@zomco Yeah the yolo can do localization and recognition at the same time, but I think the last layer should be a little bit larger.

**Comment by zomco** [comment]
@SunAriesCN How to design the network? Is there any paper about that?

**Comment by ghost** [comment]
Does this branch work?
